File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include<math.h>
using namespace std;

int solution(int n, int k) {
    int answer = 0;
    	vector<int> arr;

	while (n != 0)
	{
		arr.push_back(n % k);
		n /= k;
	}
	
	vector<long long> nums;
	string num = "";
	for (auto it = arr.rbegin(); it != arr.rend(); it++)
	{
		if (*it != 0) num += to_string(*it);
		else
		{
			if(!num.empty())
				nums.push_back(stol(num));

			num = "";
		}
	}
	if (!num.empty()) nums.push_back(stol(num));

	for (auto n : nums)
	{
		if (n == 1) continue;
		bool chk = true;
		for (long long i = 2; i <= sqrt(n); i++)
		{
			if (n % i == 0)
			{
				chk = false;
				break;
			}
			else chk = true;
		}
		if (chk) answer++;
	}
    return answer;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        string res = "";
        while(n > 0)
        {
            res+= n % k;
            n /= k;
        }
        string ans = string.Concat(res.Reverse());

        List<long> all = new List<long>();


for(int i = 0; i<ans.Length;i++)
{
    if (ans[i] == '0') continue;
    string temp = "";
    for(int j = i; j<ans.Length;j++)
    {
        if (ans[j] == '0') break;
        else temp += ans[j];
    }
    if(long.Parse(temp) == 1) continue;
    all.Add(long.Parse(temp));
    i+= temp.Length - 1;
}


foreach(var i in all)
{
    bool chk = true;
    for(long q = 2; q<=Math.Sqrt(i);q++)
    {
        if(i % q == 0)
        {
            chk = false;
            break;
        }
    }
    if (chk) answer++;
}
        return answer;
    }
}